src/ScottPlot4/ScottPlot/NumeriConversion.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace ScottPlot
+{
+    public static class NumeriConversion
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I'm normally pretty hesitant when it comes to `AggressiveInlining`, I find that my intuition as to where I should apply it is not as good as I'd like it to be.

I'd be interested to see a benchmark with and without this attribute. Or indeed checking with a disassembler to see if it gets inlined anyways, but iirc that gets difficult because it's inlined by the JITer, not the AOT compiler.

sharplab should let you see the JITted assembly: https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcDMACR2DC2A3utmbjnACzYCyAFAJTGnkC+6bQA

Reply:
At any rate, this is a minor quibble, and feel free to overrule me.

Especially since writing benchmarks + examining assembly is perhaps not worth the effort.

Reply:
> I'm normally pretty hesitant when it comes to AggressiveInlining, I find that my intuition as to where I should apply it is not as good as I'd like it to be.

I'm inlining a single cast or a single call, using this attribute can only help.

> I'd be interested to see a benchmark with and without this attribute.

(Full plot rendering benchs are coming)

|           Type |    Method |       Mean |     Error |    StdDev | Code Size |
|--------------- |---------- |-----------:|----------:|----------:|----------:|
| Decimal |       New | 15.2704 ns | 1.3047 ns | 0.0715 ns |      86 B |
|  Double |       New |  1.4432 ns | 0.2544 ns | 0.0139 ns |      41 B |
|   Int32 |       New |  1.6816 ns | 0.1801 ns | 0.0099 ns |      47 B |
|  Single |       New |  1.4643 ns | 0.6144 ns | 0.0337 ns |      45 B |
| Decimal | NewAggInl | 13.4416 ns | 1.2293 ns | 0.0674 ns |     147 B |
|  Double | NewAggInl |  0.2426 ns | 0.0863 ns | 0.0047 ns |      20 B |
|   Int32 | NewAggInl |  0.2606 ns | 0.2102 ns | 0.0115 ns |      28 B |
|  Single | NewAggInl |  0.4824 ns | 0.7316 ns | 0.0401 ns |      24 B |
| Decimal |       Old | 14.1851 ns | 0.3508 ns | 0.0192 ns |     133 B |
|  Double |       Old | 10.7178 ns | 5.1088 ns | 0.2800 ns |     136 B |
|   Int32 |       Old | 10.3100 ns | 1.0246 ns | 0.0562 ns |     126 B |
|  Single |       Old | 10.5590 ns | 0.6282 ns | 0.0344 ns |     136 B |